### PR TITLE
Quests fixes and additions

### PR DESCRIPTION
**Pull request by SimonCeder**
Relates to #4602.
This PR adds in all the remaining "real" quests. Specifically it adds and implements:
* Conquer City State
* Give Gold
* Pledge to Protect
* Contest Culture
* Contest Faith
* Contest Technologies
* Invest
* Bully City State
* Denounce Civilization
* Spread Religion

It also fixes a couple bugs, eg city states tasking you to build wonders that are disabled by game settings.

What remains to be done:
The pseudo-quests Barbarian Invasion, and Proxy War remain to be implemented.
We should probably switch out the string comparisons to enums at some point.

### CODE REVIEW

File: android/assets/jsons/translations/template.properties [hunk 1]
```diff
@@ -134,6 +134,9 @@ Ally =
 
 [questName] (+[influenceAmount] influence) = 
 [remainingTurns] turns remaining = 
+Current leader is [civInfo] with [amount] Culture generated. = 
+Current leader is [civInfo] with [amount] Faith generated. =
```
Comment (yairm210) [inline android/assets/jsons/translations/template.properties:138]: Current leader is [civInfo] with [amount] [stat] generated. =

**Comment by yairm210** [comment]
All in all looks good!
I will wait for others to say their piece before merging :)

File: android/assets/jsons/Civ V - Vanilla/Quests.json [hunk 2]
```diff
@@ -66,35 +65,34 @@
...
+    "name": "Contest Technologies",
     "description": "The civilization with the largest number of new Technologies researched will gain a reward.",
     "type": "Global",
     "duration": 30,
     "minimumCivs": 3
   },
   {
     "name": "Invest",
-    "description": "Our people are rejoycing thanks to a tourism boom. For a certain amount of time, any Gold donation will yeld double the Influence.",
+    "description": "Our people are rejoicing thanks to a tourism boom. For a certain amount of time, any Gold donation will yield 50% extra Influence.",
```
Comment (xlenstra) [inline android/assets/jsons/Civ V - Vanilla/Quests.json:76]: Should the 50% be parameterized?

**Comment by SomeTroglodyte** [comment]
> city states tasking you to build wonders that are disabled

... will that also fix CS giving you quests to build wonders that are unique to a different nation? (mods obviously)
... no it doesn't. But - kotlin style guide: && at line end is canon, I would personally prefer it at next line beginning, but mixing the styles...

`if (leaderString != "") questTable.add(leaderString.toLabel()).row()` - so the Table sometimes ends with an Cell.endRow marker sometimes not? I suppose it doesn't matter.

Curious: getCityStateTarget has no checks for relationship levels, alliances, or at-war

No GameSpeed effect on CONTEST_LENGTH? Well, could be...

File: core/src/com/unciv/logic/civilization/PolicyManager.kt [hunk 3]
```diff
@@ -23,6 +23,7 @@ class PolicyManager {
 
     var freePolicies = 0
     var storedCulture = 0
+    var totalCulture = 0 // for contests
```
Comment (xlenstra) [inline core/src/com/unciv/logic/civilization/PolicyManager.kt:26]: I'd prefer if the name of this variable also showed it has something to do with contests, and maybe it should be saved in diplomacymanagers anyway, as it is only ever used for city state contests?

File: core/src/com/unciv/logic/civilization/QuestManager.kt [hunk 4]
```diff
@@ -37,6 +39,8 @@ class QuestManager {
 
         const val GLOBAL_QUEST_MAX_ACTIVE = 1
         const val INDIVIDUAL_QUEST_MAX_ACTIVE = 2
+
+        const val CONTEST_LENGTH = 30
```
Comment (xlenstra) [inline core/src/com/unciv/logic/civilization/QuestManager.kt:43]: Not depended on game speed? I don't remember myself, so feel free to ignore this is if it isn't

File: core/src/com/unciv/logic/civilization/QuestManager.kt [hunk 5]
```diff
@@ -358,14 +397,29 @@ class QuestManager {
 
     /** Returns the score for the [assignedQuest] */
     private fun getScoreForQuest(assignedQuest: AssignedQuest): Int {
-        @Suppress("UNUSED_VARIABLE")  // This is a work in progress
         val assignee = civInfo.gameInfo.getCivilization(assignedQuest.assignee)
         return when (assignedQuest.questName) {
-            // Waiting for contest quests
+            QuestName.ContestCulture.value -> assignee.policies.totalCulture - assignedQuest.data1.toInt()
+            QuestName.ContestFaith.value -> assignee.religionManager.totalFaith - assignedQuest.data1.toInt()
+            QuestName.ContestTech.value -> assignee.tech.getNumberOfTechsResearched() - assignedQuest.data1.toInt()
```
Comment (xlenstra) [inline core/src/com/unciv/logic/civilization/QuestManager.kt:404]: On first glance this looks like researching 'future tech' doesn't count towards this, is this correct and intentional?

File: core/src/com/unciv/logic/civilization/QuestManager.kt (hunk 4 above)
Comment (SimonCeder) [reply core/src/com/unciv/logic/civilization/QuestManager.kt:43]: It is depending on game speed, that's just an unused constant. I'll delete it.

File: android/assets/jsons/Civ V - Vanilla/Quests.json (hunk 2 above)
Comment (SimonCeder) [reply android/assets/jsons/Civ V - Vanilla/Quests.json]: sure, why not.

File: core/src/com/unciv/logic/civilization/PolicyManager.kt (hunk 3 above)
Comment (SimonCeder) [reply core/src/com/unciv/logic/civilization/PolicyManager.kt]: ok, moved it to CivilizationInfo.

File: core/src/com/unciv/logic/civilization/QuestManager.kt (hunk 5 above)
Comment (SimonCeder) [reply core/src/com/unciv/logic/civilization/QuestManager.kt:410]: Yes, correct and intentional.

**Comment by SimonCeder** [comment]
> > city states tasking you to build wonders that are disabled
> 
> ... will that also fix CS giving you quests to build wonders that are unique to a different nation? (mods obviously)
> ... no it doesn't. But - kotlin style guide: && at line end is canon, I would personally prefer it at next line beginning, but mixing the styles...

Ok, added yet another check and made it a bit prettier

> `if (leaderString != "") questTable.add(leaderString.toLabel()).row()` - so the Table sometimes ends with an Cell.endRow marker sometimes not? I suppose it doesn't matter.

The quest description was definitely added a couple lines above with a .row() so it should always end with one.

> Curious: getCityStateTarget has no checks for relationship levels, alliances, or at-war

Yes, correct. It's only proximity.

> No GameSpeed effect on CONTEST_LENGTH? Well, could be...

That's just an old unused constant, the actual code corrects for game speed. I removed the constant.

**Comment by SomeTroglodyte** [comment]
> row

Blindness on my part. I blame the smaller screen... :ghost:

**Comment by logicminimal** [comment]
does this implement different kinds of quests given by city state personality? like difficult and often agressive quests from agressive city states etc

**Pull request merged**